app/vmselect/promql/rollup_test.go

```diff
@@ -117,31 +117,40 @@ func TestRollupIderivDuplicateTimestamps(t *testing.T) {
 }
 
 func TestRemoveCounterResets(t *testing.T) {
-	removeCounterResets(nil)
+	removeCounterResets(nil, nil, 0)
 
 	values := append([]float64{}, testValues...)
-	removeCounterResets(values)
+	timestamps := append([]int64{}, testTimestamps...)
+	removeCounterResets(values, timestamps, 0)
 	valuesExpected := []float64{123, 157, 167, 188, 221, 255, 320, 332, 364, 396, 398, 398}
 	testRowsEqual(t, values, testTimestamps, valuesExpected, testTimestamps)
 
 	// removeCounterResets doesn't expect negative values, so it doesn't work properly with them.
 	values = []float64{-100, -200, -300, -400}
-	removeCounterResets(values)
-	valuesExpected = []float64{-100, -100, -100, -100}
 	timestampsExpected := []int64{0, 1, 2, 3}
+	removeCounterResets(values, timestampsExpected, 0)
+	valuesExpected = []float64{-100, -100, -100, -100}
 	testRowsEqual(t, values, timestampsExpected, valuesExpected, timestampsExpected)
 
 	// verify how partial counter reset is handled.
 	// See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2787
 	values = []float64{100, 95, 120, 119, 139, 50}
-	removeCounterResets(values)
-	valuesExpected = []float64{100, 100, 125, 125, 145, 195}
 	timestampsExpected = []int64{0, 1, 2, 3, 4, 5}
+	removeCounterResets(values, timestampsExpected, 0)
+	valuesExpected = []float64{100, 100, 125, 125, 145, 195}
 	testRowsEqual(t, values, timestampsExpected, valuesExpected, timestampsExpected)
 
+	// verify that staleness interval is respected during resets
+	values = []float64{10, 12, 14, 4, 6, 8, 6, 8, 4, 6}
+	timestamps = []int64{10, 20, 30, 60, 70, 80, 90, 100, 120, 130}
+	valuesExpected = []float64{10, 12, 14, 4, 6, 8, 14, 16, 4, 6}
```

Comment:
>respect staleness detection in increase, increase_pure and delta functions when time series has gaps and `-search.maxStalenessInterval` is set.

I don't understand the difference between returning
```
valuesExpected = []float64{10, 12, 14, 4, 6, 8, 14, 16, 4, 6}
```
and 
```
valuesExpected = []float64{10, 12, 14, 18, 20, 22, 28, 30, 34, 36}
```
in terms of the results from `increase_pure` and `increase`, both of them yield the same final results, no?

Reply:
I tested with above samples
```
"values":
[10,12,14,4,6,8,6,8,4,6]
"timestamps":[1737112767000,1737112777000,1737112787000,1737112817000,1737112827000,1737112837000,1737112847000,1737112857000,1737112877000,1737112887000]
```
setting `-search.maxStalenessInterval=5m` gives:
<img width="1451" alt="image" src="https://github.com/user-attachments/assets/3b592370-ecd2-4e30-abeb-74815e3ceb7f" />

setting `-search.maxStalenessInterval=10s` gives:
![image](https://github.com/user-attachments/assets/399a0e12-783b-41b6-91da-08e6f257ac43)

Reply:
> I tested with above samples on this fix branch:

did you try the same tests on master branch?

Reply:
I did, it's the same and caches are all disabled.